app/models/course.rb

```diff
@@ -5,6 +5,14 @@ class Course < ActiveRecord::Base
                                OR lower(title) like ?",
                                "%#{q.to_s.downcase.gsub(/\s+/,'')}%", "%#{q.to_s.downcase}%") }
 
+  # Get all prereq info from the database.
+  # Return an array of arrays of courses.
+  def prerequisites
+    Prerequisite.where(postrequisite: self).reduce([]) do |arr, prereq|
+      arr << Course.where('id IN (?)', prereq.prerequisite_ids)
+    end
+  end
```

Comment:
http://stackoverflow.com/questions/17012330/rails-postgres-foreign-keys-stored-in-array-to-create-1-many-association